lib/graphql/query.rb

```diff
@@ -265,6 +266,21 @@ def sanitized_query_string
       }
     end
 
+    # @return [String] An opaque hash identifying this query-variables combination
+    def fingerprint
+      @fingerprint ||= "#{query_fingerprint}/#{variables_fingerprint}"
+    end
+
+    # @return [String] An opaque hash for identifying this query's given query string
+    def query_fingerprint
+      @query_hash ||= Fingerprint.generate(query_string)
```

Comment:
Is it worth fingerprinting a version of the query string with normalized whitespace formatting?

Reply:
We do `.strip` the query before we hash it. Normalizing all whitespace would be expensive since you'd have to parse/reprint the query to avoid mangling string constants.